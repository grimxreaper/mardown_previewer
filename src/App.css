body {
  background-color: #1dd1a1;
}

.preview {
  background-color: white;
}

textarea, .preview {
  min-height: 400px;
}

.preview img {
  max-width: 100%;
}

h1 {
  color: rgb(37, 33, 33);
  text-shadow: 1px 1px 4px #b18be6;
  margin-bottom: 20px;
}

h5, h1 {
  font-family: 'Baloo 2', cursive;
}

.preview blockquote {
  font-family: 'Baloo 2', cursive;
}

blockquote p:last-child {
  font-size: 17px;
}

.preview li {
  font-size: 12px;
}

h5 {
  font-size: 18px;
  margin-bottom: 8px;
}

p {
  font-size: 13px;
}