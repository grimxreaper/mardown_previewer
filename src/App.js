import './App.css';

class App extends React.Component {
  constructor(props) {
    super(props)

    state = {
      edit: true
    }
  }

handleChange = () => {
  //when text is entered into the editor element
  this.setState(() => ({
    edit: !state.edit //update preview element
    
  }))
}



  render() {
    <div>
      <textarea
        id="editor"

        //trigger a change event here, and update the state of preview to render
        //what was written in the editor as html

      ></textarea>
      {/* When I enter text into the #editor element, the #preview element
is updated as I type to display the content of the textarea. */}

      {/* When I enter GitHub flavored markdown into the #editor element,
the text is rendered as HTML in the #preview element as I type. */}
      <div
        id="preview"
      >
        {//result of preview
        }

;
      </div>
    </div>
  }
}




export default App;
