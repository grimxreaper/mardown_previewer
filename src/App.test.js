// import { render, screen, fireEvent } from "@testing-library/react";
import App from "./App";
import userEvent from "@testing-library/user-event";

import React from "react";
// import { StateMock } from '@react-mock/state';
import { render, screen, fireEvent, getByLabelText } from "@testing-library/react";

//snapshot
it("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });


//a test for entering text a certain way in the editor using markdown
//and that appearing in the previewer in the correct way

//click and type text in the editor

test('contains heading 1', () => {
  render(<App />);

    expect(screen.getByRole('article')).toHaveTextContent("Heading 1")
})

// test('renders heading 1 correctly', () => {
//   const { getByTestID } = render(<App />);
//   const input = screen.getByLabelText('editor');
//   const previewer = screen.getByLabelText('previewer')
  
//   fireEvent.change(input, { target: { value: '#Testing Another Heading 1'}});
//   expect(previewer.value).toBe('#Testing Another Heading 1');

//   // expect(screen.getByRole('article')).toHaveTextContent("Heading 1");

// })

test('renders heading 1 correctly', () => {
  const { getByTestID } = render(<App />);
  // const originalPreview = screen.getByLabelText('previewer').toHaveTextContent("This is a paragraph This is bolded text Heading 1 Heading 2 choose a color here This is an inline <div></div> This is a block of code: let z = 'zahra'; let y = 'yael'; let f = z + y; strawberry mochi heart filled macaroons coconut water When you give up, your dreams and everything else, they’re gone. Hard work is worthless for those that don’t believe in themselves. – Uzumaki Naruto");
  // const previewer = screen.getByLabelText('previewer')
  const input = screen.getByLabelText('editor');
  
  fireEvent.change(input, { target: { value: '# Testing Another Heading 1'}});
  expect(screen.getByRole('article')).toHaveTextContent('Testing Another Heading 1');

  // expect(screen.getByRole('article')).toHaveTextContent("Heading 1");

})