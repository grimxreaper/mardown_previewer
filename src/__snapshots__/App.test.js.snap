// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      class="text-center mt-4"
    >
      Convert Your Markdown!
    </h1>
    <div
      class="row"
    >
      <div
        class="col-6"
      >
        <h5
          class="text-center"
        >
          Enter your markdown here:
        </h5>
        <textarea
          aria-label="editor"
          class="form-control p-2"
          id="editor"
        >
            This is a paragraph
  **This is bolded text**

  # Heading 1
  ## Heading 2

  [choose a color here](https://flatuicolors.com/palette/ca)

  This is an inline \`&lt;div&gt;&lt;/div&gt;\`
  This is a block of code:

  \`\`\`
    let z = 'zahra';
    let y = 'yael';
    let f = z + y;
  \`\`\`

  - strawberry mochi
  - heart filled macaroons
    - coconut water

  ![Image](https://i.pinimg.com/564x/83/7c/c3/837cc3efb59ffe05d7b0ae7f3177151b.jpg)

  &gt; When you give up, your dreams and everything else, they’re gone. Hard work is worthless for those that don’t believe in themselves. – Uzumaki Naruto

        </textarea>
      </div>
      <div
        class="col-6"
      >
        <h5
          class="text-center"
        >
          See the result:
        </h5>
        <div
          aria-label="previewer"
          class="preview rounded p-2"
          id="preview"
          role="article"
        >
          <p>
              This is a paragraph
            <br />
              
            <strong>
              This is bolded text
            </strong>
          </p>
          

          <h1
            id="heading-1"
          >
            Heading 1
          </h1>
          

          <h2
            id="heading-2"
          >
            Heading 2
          </h2>
          

          <p>
              
            <a
              href="https://flatuicolors.com/palette/ca"
            >
              choose a color here
            </a>
          </p>
          

          <p>
              This is an inline 
            <code>
              &lt;div&gt;&lt;/div&gt;
            </code>
            <br />
              This is a block of code:
          </p>
          

          <pre>
            <code>
                let z = 'zahra';
  let y = 'yael';
  let f = z + y;

            </code>
          </pre>
          

          <ul>
            

            <li>
              strawberry mochi
            </li>
            

            <li>
              heart filled macaroons
              <ul>
                

                <li>
                  coconut water
                </li>
                

              </ul>
              

            </li>
            

          </ul>
          

          <p>
              
            <img
              alt="Image"
              src="https://i.pinimg.com/564x/83/7c/c3/837cc3efb59ffe05d7b0ae7f3177151b.jpg"
            />
          </p>
          

          <blockquote>
            

            <p>
              When you give up, your dreams and everything else, they’re gone. Hard work is worthless for those that don’t believe in themselves. – Uzumaki Naruto
            </p>
            

          </blockquote>
          

        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
